### PR TITLE
Parameterize docker registry URL

### PR DESCRIPTION
**Pull request by irregulator**
This will allow Jenkins to swap the variable accordingly.

### CODE REVIEW

**Comment by irregulator** [comment]
[sc-61884]

File: Jenkinsfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ node {
       updateGithubCommitStatus('PENDING', "${env.WORKSPACE}/src")
 
       stage('Build debian package') {
-	sh 'make jammy_package'
+	sh 'make jammy_package EXOSCALE_DOCKER_REGISTRY=${EXOSCALE_DOCKER_REGISTRY}'
```
Comment (xgattuso) [inline Jenkinsfile]:
```suggestion
	sh 'make jammy_package EXOSCALE_DOCKER_REGISTRY=${EXOSCALE_DOCKER_REGISTRY}'
```

Shouldn't be prepended?

Comment (irregulator) [reply Jenkinsfile]: I think you're right; but somehow it worked here: https://ci.internal.exoscale.ch/blue/rest/organizations/jenkins/pipelines/puppet-omnibus/runs/6/nodes/17/steps/18/log/?start=0

Comment (exo-cedric) [reply Jenkinsfile]: Agree with @xgattuso ; if it works, it's confusing :wink:

File: Makefile [hunk 2]
```diff
@@ -3,6 +3,7 @@ VERSION       := $(or ${puppet_version},"3.8.5")
 ITERATION     := $(or ${puppet_vendor_version},y7)
 PACKAGE_NAME  := "puppet-omnibus"
 PKG_ITERATION := exo5
+DOCKER_REGISTRY := $(or ${EXOSCALE_DOCKER_REGISTRY}, "registry.internal.exoscale.ch")
```
Comment (exo-cedric) [inline Makefile]:
```suggestion
EXOSCALE_DOCKER_REGISTRY ?= "registry.internal.exoscale.ch"
```

File: Makefile [hunk 3]
```diff
@@ -46,7 +47,7 @@ puppet_exo:
 
 docker: require_os
 	flock /tmp/puppet_omnibus_$(OS)_docker_build.lock \
-	docker build -f Dockerfile.$(OS) -t package_puppet_omnibus_$(OS) .
+	docker build -f Dockerfile.$(OS) -t package_puppet_omnibus_$(OS) --build-arg EXOSCALE_DOCKER_REGISTRY=$(DOCKER_REGISTRY) .
```
Comment (exo-cedric) [inline Makefile]:
```suggestion
	docker build -f Dockerfile.$(OS) -t package_puppet_omnibus_$(OS) --build-arg EXOSCALE_DOCKER_REGISTRY=$(EXOSCALE_DOCKER_REGISTRY) .
```

**Review by exo-cedric — APPROVED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (irregulator) [reply Jenkinsfile]: Well, order doesn't play a role it seems.

```
➜  russell src/pkg/puppet-omnibus git:(fix_docker_registry) ✗ cat Makefile
.PHONY: my_target

my_target: $(info $(MYVAR))
        @:

➜  russell src/pkg/puppet-omnibus git:(fix_docker_registry) ✗ make MYVAR=la my_target 
la
➜  russell src/pkg/puppet-omnibus git:(fix_docker_registry) ✗ make my_target MYVAR=ko
ko
```

Will change it as you suggested, for the sake of clarity.

Comment (irregulator) [reply Jenkinsfile]: https://github.com/exoscale/puppet-omnibus/pull/2/commits/d864ed7a7e49bae00ce32dcb1d2b3cec07ebb324

**Pull request merged**